packages/block-editor/src/components/date-format-control/index.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * External dependencies
+ */
+import { uniq } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { _x, sprintf, __ } from '@wordpress/i18n';
+import { dateI18n } from '@wordpress/date';
+import { useState, createInterpolateElement } from '@wordpress/element';
+import {
+	SelectControl,
+	TextControl,
+	ExternalLink,
+} from '@wordpress/components';
+
+/**
+ * So that we can illustrate the different formats in the dropdown properly,
+ * show a date that has a day greater than 12 and a month with more than three
+ * letters. Here we're using 2022-01-25 which is when WordPress 5.9 was
+ * released.
+ */
+const EXAMPLE_DATE = new Date( 2022, 0, 25 );
+
+/**
+ * The `DateFormatControl` component renders a dropdown that lets the user
+ * choose a _date format_. That is, how they want their dates to be formatted.
+ *
+ * @see https://github.com/WordPress/gutenberg/blob/HEAD/packages/block-editor/src/components/date-format-control/README.md
+ *
+ * @param {Object}                     props
+ * @param {string|null}                props.format     The selected date
+ *                                                      format.
+ * @param {string}                     props.siteFormat The site's date format,
+ *                                                      used to show what the
+ *                                                      date will look like if
+ *                                                      user selects _Site
+ *                                                      default_.
+ * @param {( format: string ) => void} props.onChange   Called when a selection
+ *                                                      is made.
+ */
+export default function DateFormatControl( { format, siteFormat, onChange } ) {
+	// Suggest a short format, medium format, long format, and a standardised
+	// (YYYY-MM-DD) format. The short, medium, and long formats are localised as
+	// different languages have different ways of writing these. For example, 'F
+	// j, Y' (April 20, 2022) in American English (en_US) is 'j. F Y' (20. April
+	// 2022) in German (de). The resultant array is de-duplicated as some
+	// languages will use the same format string for short, medium, and long
+	// formats.
+	const suggestedFormats = uniq( [
```

Comment:
Not a blocker for now, but in the future we might want to allow an extra param on this component to either replace or append to this list of suggested formats. I reckon we could leave that to a follow-up if and when we need it, though!

Also, just double-checking, is the `uniq` call here just in case the translated / localized strings winds up with duplicates?

Reply:
Yeah, I'm not sure how best to make this component "generic" so I figure keep the API minimal for now and it can evolve with time.

Correct. I anticipate that some languages don't have e.g. a distinction between medium dates (Jan 5) and long dates (January 5) and so best to de-dupe the list of localised strings. At the very least, I know most Asian languages use YYYY-MM-DD which is a format we already have by default.

Reply:
> Also, just double-checking, is the uniq call here just in case the translated / localized strings winds up with duplicates?

I was just coming to this line to ask the same question. 😄  Is it worth documenting this inline somewhere?